lib/repository/i_repository.dart

```diff
@@ -0,0 +1,3 @@
+abstract class IRepository{
+  Future<List> readJsonData();
```

Comment:
I'd rename this method as `readData`, `loadData`, `getData` or whatever you want, but a generic name, so that you're not tied to strictly read a Json data by the naming.